### PR TITLE
Fix/simple table collapse

### PR DESCRIPTION
**Pull request by nick-lanners**
## Description

This PR adjusts some functionality for the SimpleTable, especially around the Collapsible feature:
- handle items with no collapsible content better with CSS
- fix onRowClick functionality with collapsible
- lift collapsed items state out of the component so consumers have more flexibility with how it works
 - ie make network calls or watch state for style changes
- write tests to handle these changes 

## Todo

- [x] Bump version and add tag

### CODE REVIEW

File: src/components/simple-table/__stories__/SimpleTable.stories.tsx [hunk 1]
```diff
@@ -618,3 +639,63 @@ export const Custom = () => {
...
+        collapsible={{
+          collapseContent,
+          enabled: true,
+          expandedItemKeys,
+          toggleItem,
+        }}
+        headers={tableHeaders}
+        items={items}
+        keyExtractor={item => String(item.id)}
+        onRowClick={item => alert(`Clicked ${item.name}`)}
```
Comment (goodpickle) [inline src/components/simple-table/__stories__/SimpleTable.stories.tsx:697]: Why the alert?

**Comment by ckgardner** [comment]
Screenshot 2024-12-04 at 11 05 09 AM
 The expanded content spacing seems a little off

**Comment by nick-lanners** [comment]
> Screenshot 2024-12-04 at 11 05 09 AM The expanded content spacing seems a little off (almost touching the border)

@ckgardner The content for the collapse is controlled by the consumer, so the css would be controlled on that end. I have designs that don't have any padding and such in the collapse: https://www.figma.com/design/sq4q6XA4CjrQWSdnh3vnXM/Billing?node-id=838-97897&m=dev. I'm not too concerned that the stories look pretty, as long as they show the functionality correctly.

**Comment by ckgardner** [comment]
> > Screenshot 2024-12-04 at 11 05 09 AM The expanded content spacing seems a little off (almost touching the border)
> 
> @ckgardner The content for the collapse is controlled by the consumer, so the css would be controlled on that end. I have designs that don't have any padding and such in the collapse: https://www.figma.com/design/sq4q6XA4CjrQWSdnh3vnXM/Billing?node-id=838-97897&m=dev. I'm not too concerned that the stories look pretty, as long as they show the functionality correctly.

I still think it's worthwhile to have good looking stories to reflect how we want things to look 🤷🏼 , but I get what you are saying with it not being critical.

**Review by ckgardner — DISMISSED** [review]

**Review by Thienhuynh95 — APPROVED** [review]

**Pull request merged**